### ISSUE TITLE
OpenCV.js, build with ORB and features2d

### ISSUE THREAD

**Issue by Ufosek** [opencv/opencv#11300]
I am trying to build a simple web/js app that would detect image on camera. I have already done this on iOS and Android using OpenCV so my obvious choice was to use OpenCV.js (latest release).

I built opencv.js as in tutorial https://docs.opencv.org/3.4.1/d4/da1/tutorial_js_setup.html and created simple app that changes colorspace from camera to grayscale and shows it on canvas.

However I don't know how to use ORB or it is just not included in opencv.js build. When I try to do as follows: `let orb = new cv.ORB()` I got

> cv.ORB is not a constructor

I also tried to use it in different way like in python cv.ORB_create but with the same result. There is no tutorial on docs.opencv with this. I am not sure whether it should be allocated in different way or it is missing in build.

I enabled these options in opencv/platforms/js/build_js.py:

> "-DBUILD_opencv_calib3d=On","-DBUILD_opencv_dnn=ON","-DBUILD_opencv_features2d=ON","-DBUILD_opencv_flann=On","-DBUILD_opencv_ml=On",

But with no effect. I also tried approach in comment here with the same result: [http://answers.opencv.org/question/184208/opencvjs-with-features2d/](url)

Could anyone tell me what else could I do wrong?

**Comment by rishantagarwal** [comment]
I am working with the same problem currently. I tried creating different things, like  expose orb class through the binding generator python script , but it shows bounding error. I tried making classes  for orb util, feature2d and ptr in core binding cpp and also moved the abstract class of orb _impl to hpp file , it does expose few things but still I don't have access to the instance created. Any pointers to this , I think all this is quite useful for everyone and once it's fixed I would be glad to make a PR for the same
